plugins/k8smeta/src/plugin.cpp

```diff
@@ -302,42 +302,65 @@ void my_plugin::do_initial_proc_scan()
             continue;
         }
 
-        // Example of the path: `/proc/1/cgroup`
-        proc_path = std::string(proc_root)
-                            .append("/")
-                            .append(file_name.c_str())
-                            .append("/cgroup");
+        // Now scan /proc/pid/task for threads
+        std::string task_dir = proc_root + "/" + file_name.c_str() + "/task";
```

Comment:
`std::string` + `char*` + `char*` + `char*` should result in an error, why not `file_name` instead of `file_name.c_str()`?